src/nvim/ex_cmds2.c

```diff
@@ -1339,6 +1339,22 @@ void dialog_changed(buf_T *buf, int checkall)
   }
 }
 
+/// Ask the user whether to close the terminal buffer or not.
+///
+/// @param buf The terminal buffer.
+/// @return bool Whether to close the buffer or not.
+bool dialog_close_terminal(buf_T *buf)
+{
+  char_u buff[DIALOG_MSG_SIZE];
+
+  dialog_msg(buff, _("Close \"%s\"?"),
+             (buf->b_fname != NULL) ? buf->b_fname : (char_u *)_("Untitled"));
```

Comment:
just untranslated "?" is fine instead of "Untitled" (which is not common in vim, more like MS Word).

Reply:
I took that directly from `dialog_changed()`: https://github.com/neovim/neovim/blob/9adb6ed74ac72f6a3a694cc1502f783426e22c4d/src/nvim/ex_cmds2.c#L1288-L1290

But I agree with the "?". It probably never happens that the current terminal buffer's name is NULL.